### ISSUE TITLE
[Editor]: available blocks of fields differ for objects of the same type

### ISSUE THREAD

**Issue by PetbkA** [organicmaps/organicmaps#433]
There's a couple of "churches" in Astrakhan:
- [Никольская Надвратная Церковь (4125105)](https://www.openstreetmap.org/relation/4125105)
- [Троицкий Собор (83336219)](https://www.openstreetmap.org/way/83336219)

In OM their `Category` is _Church_, in OSM `amenity` = _[place_of_worship](https://wiki.openstreetmap.org/wiki/Tag:amenity=place%20of%20worship?uselang=en)_.

When I'm editing the first, only the fields in `Address` block are available, while in the other there's also a `Name of the place` and `Details` blocks. These fields should be available in the first object too

**Comment by biodranik** [comment]
Please check/edit  data/editor.config file.

**Comment by PetbkA** [comment]
> Please check/edit  data/editor.config file.

I don't seee such file neither in _/data/data/app.omaps_ nor in _/<ExternalSD>/Android/data/app.omaps_.

I have root access. File browser: ES File Explorer Pro 1.1.4.1

**Comment by biodranik** [comment]
This file is in this repository.

**Comment by PetbkA** [comment]
> This file is in this repository.

I guess you mean [this file](https://github.com/organicmaps/organicmaps/blob/master/data/editor.config). I've checked it. I guess the lines concerning the objects on question are:
``` xml
      <field_group name="poi">
        <field_ref name="name" />
        <field_ref name="street" />
        <field_ref name="housenumber" />
        <field_ref name="level" />
        <field_ref name="opening_hours" />
        <field_ref name="phone" />
        <field_ref name="email" />
        <field_ref name="website" />
      </field_group>
...
      <type id="amenity-place_of_worship">
        <include group="poi" />
      </type>
```

So both of the objects should be treated the same way, but they aren't. As for difference between objects:
1. The first has [type](https://wiki.openstreetmap.org/wiki/Key:type) = _multipolygon_, although I can't see anything in _editor.conf_ about it;
2. The second has tag [historic](https://wiki.openstreetmap.org/wiki/Historic) = _castle_.

I doubt that the first object should have [historic](https://wiki.openstreetmap.org/wiki/Historic) = _castle_, so I guess that the problem is in [type](https://wiki.openstreetmap.org/wiki/Key:type) = _multipolygon_ .

I can't know for sure what should be fixed here so I wouldn't risk to do it myself

**Comment by biodranik** [comment]
Thanks for pointing it out! There is a bug, we'll check it.

**Comment by pastk** [comment]
Seems like names of POIs become not editable when they're mapped as a building, i.e. POI tags are added to the `building=yes` area. Likewise names of buildings are not editable either.

POI names are editable when they're mapped as points or non-building areas (like most schools, universities..).

**Comment by map-per** [comment]
Interesting observation: This bug occurs with `amenity=place_of_worship` but not with `tourism=hotel`

**Comment by PetbkA** [comment]
@map-per , can you give some examples?

**Comment by map-per** [comment]
Yes, e.g. https://www.openstreetmap.org/way/183873324 or https://www.openstreetmap.org/way/128215923

**Comment by map-per** [comment]
I found a pattern:
The bug only occurs for POIs where the type consists of three parts (and `building` type is also present). So e.g. `amenity-place_of_worship-christian` or `tourism-information-visitor_centre`. When the type consists of two parts like `amenity-restaurant` or `amenity-cafe` things work fine.

Haven't checked the code though, but maybe the editor prefers the full match with the `building` category over the partial match with e.g. `amenity-place_of_worship`.
